src/base/emu-i386/simx86/codebuf.c

```diff
@@ -0,0 +1,100 @@
+/*
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+/*
+ * Purpose: dlmalloc's memory spaces support
+ * and circumvent W^X restrictions.
+ *
+ * Author: @stsp
+ *
+ */
+#include <stddef.h>
+#include <assert.h>
+#include <sys/mman.h>
+#include "emu86.h"
+#include "misc/dlmalloc.h"
+
+#ifndef MAP_JIT
+#define MAP_JIT 0
+#endif
+#define CODEBUF_SZ (128 * 1024 * 1024)
```

Comment:
why 128MB? it's probably sufficient since we GC at 10000 nodes anyway. But what happens if it goes over?
From reading dlmalloc it'll just allocate elsewhere. Is that ok?